flang/lib/Optimizer/OpenMP/MapInfoFinalization.cpp

```diff
@@ -550,7 +552,39 @@ class MapInfoFinalizationPass
       // iterations from previous function scopes.
       localBoxAllocas.clear();
 
-      // First, walk `omp.map.info` ops to see if any record members should be
+      // First, walk `omp.map.info` ops to see if any of them have varPtrs
+      // with an underlying type of fir.char<k, ?>, i.e a character
+      // with dynamic length. If so, check if they need bounds added.
+      func->walk([&](mlir::omp::MapInfoOp op) {
+        if (!op.getBounds().empty())
+          return mlir::WalkResult::advance();
```

Comment:
Nit: Here you can use the `void` callback specialization of `Operation::walk()`, since we don't need to interrupt the walk or skip nested operations.
```suggestion
          return;
```